one-ui/src/main/java/com/marklogic/hub/oneui/controllers/EnvironmentController.java

```diff
@@ -114,6 +121,23 @@ public void onError(String commandName, Exception exception) {
         return payload;
     }
 
+    @RequestMapping(value = "/project-download", produces = "application/zip")
+    public void downloadProject(HttpServletRequest request, HttpServletResponse response) {
+        HubProject project = hubConfig.getHubProject();
+        File exportFile = project.getProjectDir().resolve("build").resolve("datahub-project.zip").toFile();
+        project.exportProject(exportFile);
+
+        response.setContentType(MediaType.APPLICATION_OCTET_STREAM_VALUE);
+        response.addHeader("Content-Disposition", "attachment; filename=" + exportFile.getName());
+        response.setContentLength((int) exportFile.length());
```

Comment:
Hmm - I see this, as I don't think this is necessary to call - at least I'm hoping it isn't, as it would force us to write the data out to a file first. But I'm pretty certain it doesn't need to be called, and thus shouldn't if we can write the zip directly to the response.